core/archive/manager.rs

```diff
@@ -372,11 +375,14 @@ mod tests {
         let config = Config::default();
         let am = ArchiveManager::new(&config);
 
-        let _m = mockito::mock("GET", "/sample_artifact.tar.gz")
+        let mut server = mockito::Server::new_async().await;
+        let mock_url = server.url().parse::<Url>().unwrap();
```

Comment:
**suggestion**. If we're parsing a String into a Url, we shouldn't `format!` and reparse later again. We can manipulate the Url or avoid parsing until we have a fully-formed string instead.